### PR TITLE
[feat][broker] PIP-264: Add OpenTelemetry metadata store stats

### PR DESCRIPTION
**Pull request by dragosvictor**
[PIP-264](https://github.com/apache/pulsar/blob/master/pip/pip-264.md)

### Motivation

Adds existing metadata store metrics (documented by https://pulsar.apache.org/docs/next/reference-metrics/#metadata-store-metrics) to the OpenTelemetry pipeline.

### Modifications

- Adds the metadata store metrics, as described in https://github.com/apache/pulsar-site/pull/922, to the collector runs.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change added tests and can be verified as follows:

  - Added metric test in `org.apache.pulsar.broker.stats.OpenTelemetryMetadataStoreStatsTest`

### Does this pull request potentially affect one of the following parts:

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [x] The metrics: As described in https://github.com/apache/pulsar-site/pull/922
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [x] `doc-required` https://github.com/apache/pulsar-site/pull/922
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/dragosvictor/pulsar/pull/36

### CODE REVIEW

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/stats/BatchMetadataStoreStats.java [hunk 1]
```diff
@@ -69,15 +75,25 @@ public BatchMetadataStoreStats(String metadataStoreName, ExecutorService executo
...
 
         this.batchOpsWaitingChild = OPS_WAITING.labels(metadataStoreName);
         this.batchExecuteTimeChild = BATCH_EXECUTE_TIME.labels(metadataStoreName);
         this.opsPerBatchChild = OPS_PER_BATCH.labels(metadataStoreName);
 
+        var meter = openTelemetry.getMeter("org.apache.pulsar");
+        var attributes = Attributes.of(MetadataStoreStats.METADATA_STORE_NAME, metadataStoreName);
+        this.batchMetadataStoreSizeCounter = meter
+                .upDownCounterBuilder(EXECUTOR_QUEUE_SIZE_METRIC_NAME)
+                .setDescription("The number of blocking operations in the metadata store executor queue")
```
Comment (lhotari) [inline pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/stats/BatchMetadataStoreStats.java:90]: typo here? blocking -> batch?

Comment (dragosvictor) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/stats/BatchMetadataStoreStats.java]: I copied it over from the current description of `pulsar_batch_metadata_store_executor_queue_size` ([ref](https://pulsar.apache.org/docs/next/reference-metrics/#metadata-store-metrics)), but I think "batch" makes more sense here, thank you!

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/oxia/OxiaMetadataStore.java [hunk 2]
```diff
@@ -68,10 +69,10 @@ public OxiaMetadataStore(AsyncOxiaClient oxia, String identity) {
     public OxiaMetadataStore(
             @NonNull String serviceAddress,
             @NonNull String namespace,
-            @NonNull MetadataStoreConfig metadataStoreConfig,
+            MetadataStoreConfig metadataStoreConfig,
```
Comment (heesung-sn) [inline pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/oxia/OxiaMetadataStore.java:72]: nit: can this be null?

**Review by heesung-sn — APPROVED** [review]

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/oxia/OxiaMetadataStore.java (hunk 2 above)
Comment (dragosvictor) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/oxia/OxiaMetadataStore.java:72]: It cannot, not from any codepaths so far. It was interfering with other null checks in the method so I removed it.

**Review by lhotari — APPROVED** [review]
LGTM

**Pull request merged**